src/libraries/System.IO.FileSystem/tests/Directory/SymbolicLinks.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Directory_SymbolicLinks : BaseSymbolicLinks
+    {
+        [Fact]
+        public void EnumerateDirectories_LinksWithCycles_ShouldNotThrow()
+        {
+            DirectoryInfo testDirectory = CreateDirectorySymbolicLinkToItself();
+
+            // Windows differentiates between dir symlinks and file symlinks
+            int expected = PlatformDetection.IsWindows ? 1 : 0;
```

Comment:
very minor nit: since `System.IO.FileSystem.Tests` does not target older TFMs, you can use `OperatingSystem.IsWindows()`

```suggestion
            int expected = OperatingSystem.IsWindows() ? 1 : 0;
```